var/spack/repos/builtin/packages/sensei/package.py

```diff
@@ -0,0 +1,111 @@
+    variant('miniapps', default=True, description='Enable the parallel 3D and oscillators miniapps')
+
+    # Paraview 5.6 depends on Python 3, but all SENSEI versions up to 2.2.1 do
+    # not support Python 3. So for now, until SENSEI 3, is available SENSEI
+    # can only work with Paraview 5.5.x
+    depends_on("paraview@5.5.0:5.5.2+python+mpi+hdf5", when="+catalyst")
+    depends_on("visit", when="+libsim")
+    depends_on("vtk", when="+libsim")
+    depends_on("vtk", when="~libsim ~catalyst")
+    depends_on("adios", when="+adios")
+    # SENSEI does not support Python 3 for now. Once SENSEI 3 is available, the
+    # line below should be modified
+    depends_on("python@:2", when="+python")
+    depends_on("py-numpy", when="+python")
+    depends_on("py-mpi4py", when="+python")
+    depends_on("swig", when="+python")
+
+    # Can have either LibSim or Catalyst, but not both
+    conflicts('+libsim', when='+catalyst')
+
+    def cmake_args(self):
+        spec = self.spec
+
+        # -Ox flags are set by default in CMake based on the build type
+        args = [
+            '-DCMAKE_CXX_STANDARD=11',
+            '-DCMAKE_C_STANDARD=11',
+            '-DCMAKE_POSITION_INDEPENDENT_CODE=ON'
+        ]
+
+        sensei_switch = 'ON' if '+sencore' in spec else 'OFF'
+        args.append('-DENABLE_SENSEI:BOOL={0}'.format(sensei_switch))
+
+        vtk_dir_needed = True
+
+        if '+catalyst' in spec:
+            args.append('-DENABLE_CATALYST:BOOL=ON')
+            args.append('-DENABLE_CATALYST_PYTHON:BOOL=ON')
+            pview_prefix = spec['paraview'].prefix
+            args.append('-DParaView_DIR:PATH={0}'.format(pview_prefix))
+            vtk_dir_needed = False
+        else:
+            args.append('-DENABLE_CATALYST:BOOL=OFF')
+
+        if '+libsim' in spec:
+            args.append('-DENABLE_LIBSIM:BOOL=ON')
+            args.append('-DVISIT_DIR:PATH={0}'.format(spec['visit'].prefix))
+            args.append('-DVTK_DIR:PATH={0}'.format(spec['vtk'].prefix))
+            vtk_dir_needed = False
+        else:
+            args.append('-DENABLE_LIBSIM:BOOL=OFF')
+
+        vtkio_switch = 'ON' if '+vtkio' in spec else 'OFF'
+        args.append('-DENABLE_VTK_IO:BOOL={0}'.format(vtkio_switch))
+
+        python_switch = 'OFF'
+        if '+python' in spec:
+            python_switch = 'ON'
+            python_path = spec['python'].command.path
+            args.append('-DPYTHON_EXECUTABLE:FILEPATH={0}'.format(python_path))
+        args.append('-DENABLE_PYTHON:BOOL={0}'.format(python_switch))
+
+        if '+adios' in spec:
+            args.append('-DENABLE_ADIOS:BOOL=ON')
+            args.append('-DADIOS_DIR:PATH={0}'.format(spec['adios'].prefix))
+        else:
+            args.append('-DENABLE_ADIOS:BOOL=OFF')
+
+        if vtk_dir_needed:
+            args.append('-DVTK_DIR:PATH={0}'.format(spec['vtk'].prefix))
+
+        if '+miniapps' in spec:
+            args.append('-DENABLE_PARALLEL3D:BOOL=ON')
```

Comment:
You can simplify this cascade as follows:
```py
args.extend([
    '-DENABLE_PARALLEL3D:BOOL={0}'.format(
        'ON' if '+miniapps' in spec else 'OFF'),
    '-DENABLE_OSCILLATORS:BOOL={0}'.format(
        'ON' if '+miniapps' in spec else 'OFF')
])
```

Reply:
Actually, SENSEI 3 is already out and it supports Python 3+. So I will add that to the recipe, that is SENSEI 1 and SENSEI 2 will use Python 2 and SENSEI 3 will require Python 3+.